src/v/iceberg/tests/manifest_io_test.cc

```diff
@@ -75,15 +80,90 @@ TEST_F(ManifestIOTest, TestRoundtrip) {
     ASSERT_EQ(0, m_roundtrip.entries.size());
 }
 
+TEST_F(ManifestIOTest, TestManifestListRoundtrip) {
+    manifest_list m;
+    for (int i = 0; i < 1024; i++) {
+        manifest_file file;
+        file.manifest_path = "path/to/file";
+        file.partition_spec_id = partition_spec::id_t{1};
+        file.content = manifest_file_content::data;
+        file.seq_number = sequence_number{3};
+        file.min_seq_number = sequence_number{4};
+        file.added_snapshot_id = snapshot_id{5};
+        file.added_files_count = 6;
+        file.existing_files_count = 7;
+        file.deleted_files_count = 8;
+        file.added_rows_count = 9;
+        file.existing_rows_count = 10;
+        file.deleted_rows_count = 11;
+        m.files.emplace_back(std::move(file));
+    }
+
+    // Missing manifest list.
+    auto io = manifest_io(remote(), bucket_name);
+    auto test_path = manifest_path{"foo/bar/baz"};
+    auto dl_res = io.download_manifest_list(test_path).get();
+    ASSERT_TRUE(dl_res.has_error());
+    ASSERT_EQ(dl_res.error(), manifest_io::errc::failed);
+
+    // Now upload, then try again with success.
+    auto ul_err = io.upload_manifest_list(test_path, m).get();
+    ASSERT_FALSE(ul_err.has_value());
+
+    dl_res = io.download_manifest_list(test_path).get();
+    ASSERT_FALSE(dl_res.has_error());
+    const auto& m_roundtrip = dl_res.value();
+    ASSERT_EQ(m, m_roundtrip);
+    ASSERT_EQ(m_roundtrip.files.size(), 1024);
```

Comment:
does it make sense to also check the content of the m_roundtrip?

Reply:
I was going to ask the reverse. Isn't this line redundant after the equality check on 116?

Reply:
Good point. Left a comment in the other test about why I'm checking specific fields (no equality operator), and removed this comparison here